### PR TITLE
Use seat.fail(error) to report a problem

### LINKED ISSUE #3026: Vault code is misusing `seat.exit`, so too-small loans cause opaque error
The Vault code is misusing `seat.exit` in the code path which prohibits tiny loans:

https://github.com/Agoric/agoric-sdk/blob/d286fbde334433a27ac21709797e6c10cd7f8599/packages/treasury/src/vault.js#L385-L388

According to @dtribble / @erights , instead of `throw seat.exit(string)`, that should be `throw seat.fail(error)`. `seat.exit(string)` is for normal completion, and returns `undefined`. Throwing an Error is for errorful completion.

Throwing the `undefined` returned by `seat.exit` has the super bonus bug of causing `zcfSeat.fail` to substitute its own Error (whose stack trace points at seat.js, not vault.js):

https://github.com/Agoric/agoric-sdk/blob/d286fbde334433a27ac21709797e6c10cd7f8599/packages/zoe/src/contractFacet/zcfSeat.js#L142-L146

which makes diagnosis especially challenging.

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
fixes #3026

### CODE REVIEW

File: packages/treasury/src/vault.js [hunk 1]
```diff
@@ -384,7 +384,9 @@ export function makeVaultKit(
 
     const fee = multiplyBy(wantedRun, manager.getLoanFee());
     if (amountMath.isEmpty(fee)) {
-      throw seat.exit('loan requested is too small; cannot accrue interest');
+      throw seat.fail(
```
Comment (katelynsills) [inline packages/treasury/src/vault.js:389]: This looks good, however, the `seat` type is currently any. Can we add a `@type {OfferHandler}` to the offerHandler to make sure that the linting checks that the seat methods are called correctly?

**Review by katelynsills — APPROVED** [review]
Looks good. One request for fixing the types.

**Review by warner — APPROVED** [review]

File: packages/treasury/src/vault.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/treasury/src/vault.js:389]: done (twice)

**Comment by Chris-Hibbert** [comment]
The fix to the failing test is coming in #3033.

**Pull request merged**